pygeo/geo_utils/topology.py

```diff
@@ -0,0 +1,1485 @@
+import numpy as np
+import sys
+from .norm import eDist
+from .node_edge_face import (
+    Edge,
+    setNodeValue,
+    setEdgeValue,
+    setFaceValue,
+    EdgeCmpObject,
+    nodesFromEdge,
+    nodesFromFace,
+    FaceCmpObject,
+)
+from .orientation import edgeOrientation, faceOrientation
+from .remove_duplicates import unique, uniqueIndex, pointReduce
+from .index_position import indexPosition1D, indexPosition2D, indexPosition3D
+
+# --------------------------------------------------------------
+#                Topology classes
+# --------------------------------------------------------------
```

Comment:
Should we move this file to the pygeo level? It isn't really a utility since it is just defining base classes that are used throughout.

Reply:
Good point, this appears to be just base classes used by other modules. Let's move it one level up then, and also update the imports/init for that.

Reply:
I'll move that and fix the imports, it looks like one class is used by pyLayout so there will need to be a PR fixing that import too.

Reply:
Good call, I thought I searched our codes but apparently GitHub's search is inferior to grep. I do agree with @joanibal though, `geo_utils.Topology()` doesn't make a lot of sense.